lib/src/facts/external/execution_resolver.cc

```diff
@@ -28,18 +28,39 @@ namespace facter { namespace facts { namespace external {
 
         try
         {
-            execution::each_line(path, [&facts](string const& line) {
-                auto pos = line.find('=');
-                if (pos == string::npos) {
-                    LOG_DEBUG("ignoring line in output: %1%", line);
+            string error;
+            execution::each_line(
+                path,
+                [&](string const& line) {
+                    auto pos = line.find('=');
+                    if (pos == string::npos) {
+                        LOG_DEBUG("ignoring line in output: %1%", line);
+                        return true;
+                    }
+                    // Add as a string fact
+                    string fact = line.substr(0, pos);
+                    boost::to_lower(fact);
+                    facts.add(move(fact), make_value<string_value>(line.substr(pos+1)));
                     return true;
-                }
-                // Add as a string fact
-                string fact = line.substr(0, pos);
-                boost::to_lower(fact);
-                facts.add(move(fact), make_value<string_value>(line.substr(pos+1)));
-                return true;
-            }, { execution_options::defaults, execution_options::throw_on_failure });
+                },
+                [&](string const& line) {
+                    if (!error.empty()) {
+                        error += "\n";
+                    }
+                    error += line;
+                    return true;
+                },
+                0,  // REVIEW: should we have a timeout for executable external facts?
```

Comment:
Yeah, it would need to be on the command line and affect all executable external facts (and perhaps just be a default timeout for all command executions, external fact or not).  May be something valuable to revisit in the future.